llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -226,7 +227,8 @@ static void ComputePTXValueVTs(const TargetLowering &TLI, const DataLayout &DL,
         }
         NumElts /= 2;
       } else if (EltVT.getSimpleVT() == MVT::i8 &&
-                 (NumElts % 4 == 0 || NumElts == 3)) {
+                 ((NumElts % 4 == 0 && isPowerOf2_32(NumElts)) ||
```

Comment:
This condition is rather puzzling. 
AFAICT, previously, we'd accept i8 vectors with multiples of 4 elements, and a special case of v3i8, and lowered them all as N * v4i8.

Now we'll only accept multiples of 4, that are also a power of 2 and it's not clear why. Is there any reason we should  be able to handle `v16i8` here but not `v12i8`?.

This, at the very least, needs a comment explaining what's going on, or, possibly, a change to the condition to better reflect what we're checking for here.

Reply:
> AFAICT, previously, we'd accept i8 vectors with multiples of 4 elements, and a special case of v3i8, and lowered them all as N * v4i8.

Without this PR, vector types like `v12i8` will result in a [compiler crash](https://godbolt.org/z/8KhvahMob). 

This PR should only disable this code on inputs, which, when invoked via `LowerReturn()`, would've resulted in a crash.

> This, at the very least, needs a comment explaining what's going on

I've added a comment in 0c7cdb6aaec65c9c2e75cef7bd10988a4813ed81.